src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -76,16 +76,19 @@ private static boolean isNotSameRequester(ConsentArtefact consentDetail, String
                 !consentDetail.getPatient().getId().equals(requesterId);
     }
 
-    public Mono<String> askForConsent(RequestedDetail requestedDetail) {
-        final String requestId = UUID.randomUUID().toString();
-        return validatePatient(requestedDetail.getPatient().getId())
-                .then(validateHIPAndHIU(requestedDetail))
-                .then(saveRequest(requestedDetail, requestId))
-                .then(postConsentRequest.broadcastConsentRequestNotification(ConsentRequest.builder()
-                        .detail(requestedDetail)
-                        .id(requestId)
-                        .build()))
-                .thenReturn(requestId);
+    public Mono<String> askForConsent(RequestedDetail requestedDetail, String requestId) {
+        final String consentRequestId = UUID.randomUUID().toString();
+        return consentRequestRepository.isRequestPresent(requestId)
```

Comment:
we can write method to return consent request from repository, do the filter and do rest would be a better approach.